File: server/src/api/types/token.js
import { GraphQLObjectType, GraphQLString } from 'graphql';


const GraphQLToken = new GraphQLObjectType({
  name: 'Token',
  description: 'This represents a Token',
  fields() {
    return {
      token: {
        type: GraphQLString,
        resolve(token) {
          return token;
        }
      }
    };
  }
});

export default GraphQLToken;

File: graphql_types/token.go
package graphql_types

import "github.com/graphql-go/graphql"

var Token = graphql.NewObject(graphql.ObjectConfig{
	Name:        "Token",
	Description: "This represents a Token",
	Fields: graphql.Fields{
		"token": &graphql.Field{
			Type: graphql.String,
			Resolve: func(p graphql.ResolveParams) (interface{}, error) {
				return p.Source, nil
			},
		},
	},
})
